platforms/documentation/docs/src/docs/userguide/authoring-builds/tasks/more_about_tasks.adoc

```diff
@@ -81,180 +69,161 @@ include::sample[dir="snippets/tasks/defineAsKotlinDelegatedProperty/groovy",file
 ====
 
-You can then create a task, passing the constructor arguments at the end of the parameter list.
+Then, you can create a task by passing the constructor arguments at the end of the parameter list:
 
-.Registering a task with constructor arguments using TaskContainer
 ====
 include::sample[dir="snippets/tasks/taskConstructorArgs-onTaskContainer/kotlin",files="build.gradle.kts[tags=on-task-container]"]
 include::sample[dir="snippets/tasks/taskConstructorArgs-onTaskContainer/groovy",files="build.gradle[tags=on-task-container]"]
 ====
 
-NOTE: It's recommended to use the <<task_configuration_avoidance.adoc#task_configuration_avoidance,Task Configuration Avoidance>> APIs to improve configuration time.
-
-In all circumstances, the values passed as constructor arguments must be non-null.
+The values passed as constructor arguments must be non-null.
 If you attempt to pass a `null` value, Gradle will throw a `NullPointerException` indicating which runtime value is `null`.
 
+TIP: The <<task_configuration_avoidance.adoc#task_configuration_avoidance,Task Configuration Avoidance>> APIs are recommended as they improve configuration time.
+
 [[sec:adding_dependencies_to_tasks]]
-== Adding dependencies to a task
+== Adding dependencies
 
-There are several ways you can define the dependencies of a task. In <<tutorial_using_tasks.adoc#sec:task_dependencies,Task dependencies>> you were introduced to defining dependencies using task names. Task names can refer to tasks in the same project as the task, or to tasks in other projects. To refer to a task in another project, you prefix the name of the task with the path of the project it belongs to. The following is an example which adds a dependency from `project-a:taskX` to `project-b:taskY`:
+There are several ways you can define the dependencies of a task.
+
+Defining dependencies using task names is the simplest way.
+Task names can refer to tasks in the same project as the task, or to tasks in other projects.
+
+To refer to a task in another project, you prefix the name of the task with the path of the project it belongs to.
+
+The following is an example which adds a dependency from `project-a:taskX` to `project-b:taskY`:
 
-.Adding dependency on task from another project
 ====
 include::sample[dir="snippets/tasks/addDependencyUsingPath/kotlin",files="project-a/build.gradle.kts[];project-b/build.gradle.kts[]"]
 include::sample[dir="snippets/tasks/addDependencyUsingPath/groovy",files="project-a/build.gradle[];project-b/build.gradle[]"]
 
-.Output of **`gradle -q taskX`**
 ----
-> gradle -q taskX
+$ gradle -q taskX
 include::{snippetsPath}/tasks/addDependencyUsingPath/tests/addDependencyUsingPath.out[]
 ----
 ====
 
-Instead of using a task name, you can define a dependency using a `TaskProvider` object, as shown in this example:
+Instead of using a task name, you can define a dependency using a `TaskProvider` object:
```

Comment:
🤔 While the two solutions are "equivalent" for tasks inside the same project, I believe only the first one is "isolated projects" compatible.
Or maybe we should indicate that cross project task dependencies are discouraged. But then the example above needs to be edited.